src/nni_operation.hpp

```diff
@@ -54,60 +57,56 @@ class NNIOperation {
       const Bitset parent_in, const Bitset child_in,
       const bool swap_which_child_clade_with_sister);
 
+  // ** Miscellaneous
+
+  bool IsValid() { return Bitset::SubsplitIsParentChildPair(parent_, child_); };
+
+  friend std::ostream &operator<<(std::ostream &os, const NNIOperation &nni) {
+    os << "{" << nni.parent_.SubsplitToString() << ", " << nni.child_.SubsplitToString()
+       << "}";
+    return os;
+  };
+
   Bitset parent_;
   Bitset child_;
 };
 
-// SetOfNNIs contain all NNI output parent/child pairs which are "adjacent" to the
+// * NNI
+// Contain all NNI output parent/child pairs which are "adjacent" to the
 // current SubsplitDAG. That is, pairs which are the result of an NNI on an input bitset
 // pair that are currently present in the SubsplitDAG.
-class SetOfNNIs {
+class NNISet {
  public:
-  SetOfNNIs() : set_(){};
-
-  friend bool operator==(const SetOfNNIs &lhs, const SetOfNNIs &rhs);
-  friend bool operator!=(const SetOfNNIs &lhs, const SetOfNNIs &rhs);
-
-  void Insert(NNIOperation nni_op);
-  void Insert(Bitset parent, Bitset child);
-  void Erase(NNIOperation nni_op);
-  void Erase(Bitset parent, Bitset child);
-  void Clear();
-  size_t GetSize() const;
-
- private:
-  std::set<NNIOperation> set_;
+  friend bool operator==(const NNISet &lhs, const NNISet &rhs) {
+    return lhs.data_ == rhs.data_;
+  };
+  friend bool operator!=(const NNISet &lhs, const NNISet &rhs) {
+    return !(lhs == rhs);
+  };
+
+  void insert(NNIOperation nni_op) { data_.insert(nni_op); };
+  void insert(Bitset parent, Bitset child) { data_.insert({parent, child}); };
+  void erase(NNIOperation nni_op) { data_.erase(nni_op); };
+  void erase(Bitset parent, Bitset child) { data_.erase({parent, child}); };
+  void clear() { data_.clear(); };
+
+  size_t size() const { return data_.size(); };
+  std::set<NNIOperation> &GetData() { return data_; };
+
+  std::set<NNIOperation>::iterator begin() const { return data_.begin(); };
+  std::set<NNIOperation>::iterator end() const { return data_.end(); };
+
+  NNIOperation at(const size_t i) const {
```

Comment:
I think that the right way to do this is https://stackoverflow.com/a/20477588/467327 ... 

Note I changed above to
```
  void insert(Bitset parent, Bitset child) { data_.insert({parent, child}); };
```

I'm still not convinced that this merits adding an NNISet object.